include/dev/dram/device.h

```diff
@@ -226,4 +328,147 @@ void dram_get_write_enable_offset_timing(dram_freq_cfg_t *dram_cfg,
 void dram_get_write_data_offset_timing(dram_freq_cfg_t *dram_cfg,
                                        uint8_t *offset);
 
+/**
+ * @brief   DRAM Prepare Command Bus Training Packet Sequence
+ *
+ * @details Creates a set of packets in order to send the necessary DRAM signals
+ *          specifying VREF setting to use and command address and clock signals
+ *          during command bus training.
+ *
+ * @note    This will always create the packets at time offset of 1 from
```

Comment:
I would probably leave an additionally note discouraging users from already having packets in the buffer. It makes it more likely you'll experience an error where you can't fit anymore packets. You could say something like "recommend only calling this if buffer is empty" or something like that.

This applies for all prepare_*_sequence

Reply:
The only prepare_*_sequence which will probably always be called after another one is the rdfifo sequence.  We can combine the wrfifo/rdfifo sequence into one if that would be cleaner (I expect them to always be used together).  Let me know your thoughts.  I will add the note to all others though as that is the intent to only be used in an empty buffer with training intention.

Reply:
You can leave them separate. Adding a note to the others is fine.

Reply:
Added a note specifying to be used with an empty buffer.